src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -142,5 +142,52 @@ internal static MsalTokenResponse CreateFromBrokerResponse(Dictionary<string, st
 
             return response;
         }
-    }
+
+        /// <remarks>
+        /// This method does not belong here - it is more tied to the Android code. However, that code is
+        /// not unit testable, and this one is. 
+        /// The values of the JSON response are based on 
+        /// https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerResult.java
+        /// </remarks>
+        internal static MsalTokenResponse CreateFromAndroidBrokerResponse(string jsonResponse, string correlationId)
+        {
+            dynamic authResult = JObject.Parse(jsonResponse);
+            var errorCode = authResult[BrokerResponseConst.BrokerErrorCode]?.ToString();
+
+            if (!string.IsNullOrEmpty(errorCode))
+            {
+                return new MsalTokenResponse
+                {
+                    Error = errorCode,
+                    ErrorDescription = authResult[BrokerResponseConst.BrokerErrorMessage]?.ToString(),
+                };
+            }
+
+            MsalTokenResponse msalTokenResponse = new MsalTokenResponse()
+            {
+                Authority = authResult[BrokerResponseConst.Authority],
+                AccessToken = authResult[BrokerResponseConst.AccessToken],
+                IdToken = authResult[BrokerResponseConst.IdToken],
+                CorrelationId = correlationId, // Android response does not expose Correlation ID
+                Scope = authResult[BrokerResponseConst.AndroidScopes], // sadly for iOS this is "scope" and for Android "scope"
```

Comment:
what's the difference between iOS and Android on the scope value?

Reply:
Sorry, bad comment. It's supposed to say "scope" vs "scopes"